R/join_keys-parents.R

```diff
@@ -78,6 +78,10 @@ parents.teal_data <- function(x) {
   new_parents <- list()
 
   for (dataset in names(value)) {
+    if (checkmate::test_scalar_na(value[[dataset]])) {
```

Comment:
It seems this could be covered by `assert_string`. Is it because the custom error message?

Reply:
Yes, but now that I look at this we can tweak with the `.var.name`

I'm changing to the assert below

``` r
dataset <-"d1"
value <- list(d1 = NA_character_)
checkmate::assert_string(value[[dataset]], .var.name = sprintf("value[[\"%s\"]]", dataset))
#> Error in eval(expr, envir, enclos): Assertion on 'value[["d1"]]' failed: May not be NA.
```

Reply:
Looks great :+1: